File: python/002.py
# Even Fibonacci Numbers

a, b, c = 1, 2, 0
f = [b]

while c < 4000000:
    c = a + b
    a, b = b, c

    if c % 2 == 0:
        f.append(c)

print(sum(f))

File: rust/002.rs
// Even Fibonacci Numbers

fn main() {
    let mut a = 1u32;
    let mut b = 1u32;
    let mut _sum = 0u32;
    
    loop {
        let c = a + b;
        a = b;
        b = c;

        if c > 4000000 {
            break;
        } else if c % 2 == 0 {
            _sum += c;
        }
    }

    println!("{}", _sum);
}
